subsys/net/ip/ipv6.c

```diff
@@ -403,6 +403,13 @@ enum net_verdict net_ipv6_input(struct net_pkt *pkt, bool is_loopback)
 		goto drop;
 	}
 
+#ifdef CONFIG_NET_PROMISCUOUS_MODE
+	if (net_ipv6_is_my_addr(&hdr->src)) {
```

Comment:
Seems like it could happen outside of promiscuous mode as well (you could trigger this bug easily with it but it probably exist anyway).

So you could get rid of the ifdef here.

@jukkar what's your opinion about this one?

Reply:
@tbursztyka Agree. In general, we should always drop packets with our src as it has no sense to spend resources for packet handling.

Reply:
I don't think this is the right approach. There's nothing wrong in sending a packet to our own IP address (promiscuous or not):
https://github.com/zephyrproject-rtos/zephyr/blob/4783b7ebde9252ebdf863d35e7c7e26cfd2f4dd6/subsys/net/ip/net_core.c#L307
In this solution such packets will likely get filtered out.

Reply:
@rlubos oh yes of course, if we have 2+ addresses. Completely overlooked that.

Reply:
Probably we need some extra filtering in `ipv6_nbr.c` layer instead, @jukkar?

Reply:
True. Will investigate this issue deeper